src/game/path.service.js

```diff
@@ -58,28 +58,39 @@ angular.module('ngGo.Game.Path.Service', [
       return;
     }
 
+    //Decrement move
+    this.move--;
+
     //Delete path choice
-    if (this.path[this.move]) {
+    if (typeof this.path[this.move] !== 'undefined') {
+      var branch = this.path[this.move];
       delete this.path[this.move];
-      this.branches--;
+      if (branch > 0) {
+        this.branches--;
+      }
     }
 
-    //Decrement move
-    this.move--;
     return this;
```

Comment:
What was the reason for these changes? It seems we now save the path for every move, also non-branching ones. Was this to fix something?